src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -690,6 +690,17 @@ public virtual int Run(TextWriter consoleOutput, CancellationToken cancellationT
             }
         }
 
+        //PROTOTYPE: hook to do generation, by default does nothing
+
+        /// <summary>
+        /// Peform source generation, if the compiler supports it.
+        /// </summary>
+        /// <param name="input">The compilation before any source generation has occured.</param>
+        /// <param name="parseOptions">The <see cref="ParseOptions"/> to use when parsing any generated sources.</param>
+        /// <param name="additionalTexts">Any additional texts that should be passed to the generators when run.</param>
+        /// <returns>A compilation that represents the original compilation with any additional, generated texts added to it.</returns>
+        protected virtual Compilation RunGenerators(Compilation input, ParseOptions parseOptions, ImmutableArray<AdditionalText> additionalTexts) { return input; }
```

Comment:
Should this start as `private protected` until we've decided a public API?

Reply:
Changed.